docs/source/federated-types/federated-directives.mdx

```diff
@@ -0,0 +1,302 @@
+
+**Valid locations:** Field definitions
+
+```graphql {3} title="Inventory subgraph"
+type Product @key(fields: "id") {
+  id: ID!
+  name: String! @external
+  inStock: Boolean!
+}
+
+type Query {
+  outOfStockProducts: [Product!]! @provides(fields: "name")
+  discontinuedProducts: [Product!]!
+}
+```
+
+The example subgraph above _usually_ can't resolve the `Product.name` field, but it _can_ at the `Query.outOfStockProducts` query path.
+
+
+### `@provides`
+
+Indicates that a particular subgraph can resolve particular fields, but _only_ at a particular query path.
+
+**Valid locations:** Field definitions
+
+```graphql {8} title="Inventory subgraph"
+type Product @key(fields: "id") {
+  id: ID!
+  name: String! @external
+  inStock: Boolean!
+}
+
+type Query {
+  outOfStockProducts: [Product!]! @provides(fields: "name")
+  discontinuedProducts: [Product!]!
+}
+```
+
+The example subgraph above can resolve `Product.name` for products returned by `Query.outOfStockProducts` but _not_ `Query.discontinuedProducts`.
+
+> If a subgraph `@provides` a particular field:
+>
+> * The subgraph must define that field and mark it as [`@external`](#external), as shown above.
+> * That field must be marked as either [`@shareable`](#shareable) or [`@external`](#external) in _every_ subgraph that defines it.
+>
+> Otherwise, a composition error occurs.
+
+#### Arguments
+
+<table class="field-table">
+  <thead>
+    <tr>
+      <th>Name /<br/>Type</th>
+      <th>Description</th>
+    </tr>
+  </thead>
+
+<tbody>
+
+<tr class="required">
+<td>
+
+##### `fields`
+
+`String`
+</td>
+<td>
+
+**Required.** A GraphQL selection set of field names that the subgraph can resolve at this query path.
+
+Examples:
+
+* `"name"`
+* `"name address"`
+* `"... on Person { name address }"` (valid for fields that return a union or interface)
+
+</td>
+</tr>
+
+</tbody>
+</table>
+
+### `@requires`
+
+Indicates that a particular object field's resolver requires the values of _other_ fields that are resolved by _other_ subgraphs.
+
+**Valid locations:** Field definitions
+
+```graphql {5} title="Shipping subgraph"
+type Product @key(fields: "id") {
+  id: ID!
+  size: Int @external
+  weight: Int @external
+  shippingEstimate: String @requires(fields: "size weight")
+}
+```
+
+The example subgraph above resolves a `Product` object's `shippingEstimate` field, but it requires the product's `size` and `weight` to do so. These two fields are resolved by a different subgraph, so they're marked as [`@external`](#external).
+
+See also [Contributing computed entity fields](../entities-advanced/#contributing-computed-entity-fields).
+
+#### Arguments
+
+<table class="field-table">
+  <thead>
+    <tr>
+      <th>Name /<br/>Type</th>
+      <th>Description</th>
+    </tr>
+  </thead>
+
+<tbody>
+
+<tr class="required">
+<td>
+
+##### `fields`
+
+`String`
+</td>
+<td>
+
+**Required.** A comma-separated list of `@external` object fields that this field requires.
```

Comment:
Like for `@key`/`@provides` we should probably describe it as "A GraphQL selection set of field names" rather than "A comma-separated list" (nesting is supported for instance).

Regarding the "of `@external` object fields", I'm fine saying that but I'll note that:
1. `@provides` are the exact same restrictions than `@requires` on that one, so if we say it here, we should say it for `@provides` too.
2. it is technically true that a subgraph can only require/provide stuff it doesn't resolve by itself, but nesting makes it so that it's not 100% equivalent to having _all_ the fields in a require/provide be `@external`. By which I mean that if you have something like `@provides(fields: "f { x }")` (or, equivalently, `@requires(fields: "f { x }")`), the one of `f` or `x` must be `@external`, and both can be, but it's not necessary that both _are_. Anyway, unsure it's worth getting into that kind of details here, but just wanted to provide context here so we at least avoid language that is too prohibitive. That is, "`@provides`/`@requires` can only be used to provide/require fields that the subgraph does not resolve itself" is maybe better than "all fields in the a `@provides`/`@requires` must be `@external`"?.

Reply:
I believe I've cleaned up the language here correctly, but feel free to take another look to confirm!